packages/codemirror/src/editor.ts

```diff
@@ -522,7 +522,7 @@ export class CodeMirrorEditor implements CodeEditor.IEditor {
           if (node.node.firstChild !== null) {
             return true;
           }
-          if (offset >= node.from && offset <= node.to) {
+          if (!token && offset >= node.from && offset <= node.to) {
```

Comment:
Without the check `!token`, after the first matched node, we continue to iterate through the tree and then match other nodes. The returned token is from the last matched node.
In the case of `import`, the first matched token is `{value: 'import', offset: 0, type: 'import'}` but the returned value is `{value: '', offset: 6, type: '⚠'}`, which causes the cycling issue.
I suppose that the first matched node is always the correct one so the `!token` check will ignore all other matched nodes. This fix is tested with other keywords (`class`, `def`,...) and it works as expected.